python_modules/dagster/dagster/_daemon/backfill.py

```diff
@@ -103,18 +161,45 @@ def execute_backfill_jobs(
             )
 
             try:
-                if backfill.is_asset_backfill:
-                    yield from execute_asset_backfill_iteration(
-                        backfill, backfill_logger, workspace_process_context, instance
-                    )
+                if threadpool_executor:
+                    if backfill_futures is None:
+                        check.failed(
+                            "backfill_futures dict must be passed with threadpool_executor"
+                        )
+
+                    if backfill.is_asset_backfill:
+                        future = threadpool_executor.submit(
+                            execute_asset_backfill_iteration,
+                            backfill,
+                            backfill_logger,
+                            workspace_process_context,
+                            instance,
+                        )
+                    else:
+                        future = threadpool_executor.submit(
+                            execute_job_backfill_iteration,
+                            backfill,
+                            backfill_logger,
+                            workspace_process_context,
+                            debug_crash_flags,
+                            instance,
+                        )
+                    backfill_futures[backfill_id] = future
```

Comment:
unless i am missing something i think there is an important piece of missing functionality here - the "only allow at most one future for each backfill to be in flight at once until the future finishes" code: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_daemon/sensor.py#L422-L427

without this, you could potentially be running a backfill iteration for the same backfill twice at once